File: bindings/fortran/tests/rump.f90
type (double_st) function rump(x,y)
  use cadna
  implicit none
  type (double_st), intent(in) :: x,y
  rump=9*x*x*x*x-y*y*y*y+2*y*y;
end function rump

program main
  use cadna
  implicit none
  type(double_st) x,y,z
  type(double_st),external :: rump
  x = double_st(10864._c_double)
  y = double_st(18817._c_double)
  z=rump(x,y)
  if(numericalnoise(z)) then
     write(*,*) 'numerical noise'
  endif
  write(*,*) nb_significant_digit(z)
end program main

File: bindings/python/tests/rump.py
from cadna import numericalnoise, nb_significant_digit
from cadna import double_st as real
try:
    import unittest2 as unittest
except ImportError:
    import unittest

class RumpTest(unittest.TestCase):

    def test_pass(self):

        rump = lambda x,y: 9*x*x*x*x-y*y*y*y+2*y*y
        x = real(10864)
        y = real(18817)
        v = rump(x,y)
        self.assertTrue(numericalnoise(v))
        self.assertTrue(nb_significant_digit(v)==0)
        
if __name__ == '__main__':
    unittest.main()
